### ISSUE TITLE
Differentiable field-tensor observable for optimization and surrogate training

### ISSUE THREAD

**Issue by bk-squared** [bk-squared/rfx#579]
## Motivation

`rfx/optimize_objectives.py` exposes S-parameter, impedance, bandwidth and directivity objectives. All of them reduce the simulation to a small number of scalars.

For field-surrogate work the quantity of interest is the **complex field tensor itself** — E (and ideally H) sampled on a probe grid across frequencies — and we need it as a differentiable output so that

- gradient-augmented surrogate training can consume `∂(field)/∂(params)`, and
- a field-based objective (for example: minimise the worst-case internal field over illumination) can be optimised directly.

## Current state

`add_dft_plane_probe` provides the field data, and the objectives module provides differentiable scalars, but there is no documented path that treats a DFT field tensor as a differentiable observable of the parameter pytree.

## Request

A field observable usable both as an AD output and as an objective input, roughly:

```python
obs = rfx.observables.dft_field(probe, components=("ex", "ey", "ez"), freqs=freqs)
val, grad = jax.value_and_grad(lambda p: some_functional(obs(p)))(params)
```

with at least one built-in functional over it — `max` or a soft-max over space and illumination would cover the shielding case.

## Acceptance criteria

- [ ] gradient of a scalar functional of the complex DFT field tensor matches central finite differences on one geometry and one material parameter, to a documented tolerance
- [ ] documented which components/frequencies are retained on the AD tape, since that drives memory
- [ ] one worked example in `examples/`

## Relation to other issues

Complements #577 (batched forward-mode Jacobians): #577 is about how the derivatives are computed, this is about what they are taken of.

**Comment by bk-squared** [comment]
## Narrower, higher-priority form of this request

Our direction changed and this issue became the blocker, so here is a much more specific ask that would unblock us — and it is smaller than the general field-objective API above.

**What we need is the gradient of a single bilinear scalar:**

```
g(p) = Re( u^H A(p) v )
```

where `A(p)` is the (linear) map from an exterior excitation to the interior complex field for geometry `p`, and `u`, `v` are **fixed complex vectors supplied by the caller**. Concretely that is: excite with the field pattern `v`, run one forward solve, project the resulting complex interior field onto `u`, and take the real part. One scalar out.

**Why that exact form.** For worst-case-over-illumination design the objective is the leading singular value `σ₁(A(p))`, and by the envelope theorem

```
dσ₁/dp = Re( u₁^H (∂A/∂p) v₁ )   with the singular vectors u₁, v₁ held fixed
```

so the minimax problem reduces to differentiating exactly the scalar above. One forward plus one adjoint then gives the derivative with respect to **every** geometry parameter, and the cost is independent of both the parameter count and the number of illumination angles. Without it we would finite-difference a worst case over an angular sweep, which is `N_angles × N_params` solves per optimisation step instead of about two.

**Acceptance criteria for this narrower version:**
- [ ] a caller can supply a complex excitation pattern and a complex output projection over a DFT field probe, and get `Re(u^H A v)` as a differentiable scalar
- [ ] its gradient with respect to geometry parameters matches central finite differences to a documented tolerance
- [ ] documented which parts of the DFT field stay on the AD tape, since that drives memory

This subsumes what we need from the general API; the fuller version can follow. Related: #577 has dropped in priority for us — reverse mode on a scalar is what this needs, and that already exists.

**Comment by bk-squared** [comment]
Shipped in PR #619 (merge d7e9b0f). AC-by-AC accounting:

**AC1 — FD-vs-AD gradient match**: delivered on BOTH legs in `tests/test_observables_dft_field.py`: material parameter (scalar `eps_r` via `eps_override`) and geometry parameter (one topology-density pixel through `density_to_material_fields` → `eps_override`). Central finite differences, float64 comparison arithmetic, documented 5% tolerance, and the gate loops over h ∈ {0.01, 0.02, 0.04} permanently so a coincidental-rounding green shows up as h-dependent drift. Measured: `field_energy` legs 0.10–1.13% rel err; `field_softmax` at scale-matched β=1e11: material 0.000–0.038%, geometry 0.054–0.398%. (β must be scale-matched to the field magnitude — documented in the factory docstring with worked numbers; default β=1 on rfx-scale fields degenerates to mean+log(N).)

**AC2 — what stays on the AD tape**: documented in `rfx/observables.py` ("What stays on the AD tape"): one `(n_freqs, n1, n2)` complex64 (complex128 under x64) accumulator per registered plane in the scan carry; rectangular DFT window always; the uniform forward lane still carries the time-series tape (`emit_time_series=False` is NU-only). The E/H half-step stamping hazard (H recorded at t, physically t−dt/2) is documented with the one-line user-side correction for E×H* cross terms.

**AC3 — worked example**: `examples/inverse_design/field_observable_shielding.py` — normal-incidence TFSF illumination, two internal DFT planes pooled by `field_softmax` into a worst-case-leakage scalar, minimized via `forward(eps_override)` + `jax.grad` (39.5% reduction in 12 steps; ring-down settling witness −49.5 dB vs the −40 dB bar).

Deltas vs the issue sketch, stated for the record: the observable is a result-accessor factory in the `optimize_objectives` style (the `obs(params)` composition is shown in the example as one lambda); `add_dft_plane_probe` is single-component per plane, so `components=("ex","ey","ez")` means registering three planes (pattern shown in the example); the built-in soft-max pools over space+frequency — pooling over *illumination* requires batched TFSF angles, which is addressed (and honestly declined) in #578.

Also landed with this arc: `run(distributed=True)`/`forward(distributed=True)` with registered DFT planes now fail loud instead of silently returning no plane data (behaviour change, CHANGELOG-noted).
